frb_codegen/src/generator/rust/ty.rs

```diff
@@ -64,10 +56,47 @@ pub trait TypeRustGeneratorTrait {
     fn imports(&self) -> Option<String> {
         None
     }
+
+    fn get_context(&self) -> &TypeGeneratorContext;
+
+    fn get_shared_mod_name_if_type_shared(&self, ty: &IrType) -> Option<String> {
+        if self.get_context().ir_file.is_type_shared_by_safe_ident(ty) {
+            return SHARED_MODULE.with(|data| {
+                let cloned = data.borrow().clone();
+                if cloned.is_none() {
+                    panic!("in instance in charge of `{}`: checking shared for type \"{:?}\", it is shared indeed, but the shared module name is None",
+                        self.get_context().type_name, ty
+                    );
+                }
+                cloned
+            });
+        }
+        None
+    }
+
+    fn get_shared_module_of_a_type(&self, ir_type: &IrType) -> Option<String> {
+        match ir_type {
+            IrType::Optional(inner_type) => {
+                self.get_shared_mod_name_if_type_shared(&inner_type.inner)
+            }
```

Comment:
nit: is it possible we do not make a special treatment for Optional? (answer can surely be no!)

Reply:
Logically, I think it is essential to get the inner type from an Optional Type. But I do realize that `get_shared_mod_name_if_type_shared` could be integrated into `get_shared_module_of_a_type`.  Refined in [0753dbf](https://github.com/fzyzcjy/flutter_rust_bridge/pull/1145/commits/0753dbf5a3034daecd2600ced5d06179d07e1692).